xla/service/gpu/transforms/horizontal_loop_fusion.cc

```diff
@@ -738,11 +737,22 @@ absl::StatusOr<bool> HorizontalLoopFusion::Run(
     const absl::flat_hash_set<absl::string_view>& execution_threads) {
   VLOG(2) << "Run horizontal fusion.";
 
-  // Run on the entry computation is actually enough.
-  TF_ASSIGN_OR_RETURN(bool changed,
-                      RunOnComputation(module->entry_computation()));
+  bool any_changed = false;
+  std::unique_ptr<CallGraph> call_graph = CallGraph::Build(module);
+  for (HloComputation* computation :
+       module->MakeNonfusionComputations(execution_threads)) {
+    if (computation->IsEntryComputation() ||
+        call_graph->GetComputationCallers(computation).size() == 1 &&
+            call_graph->GetComputationCallers(computation)[0]->opcode() ==
+                HloOpcode::kWhile &&
```

Comment:
Interesting, I never noticed that it only runs on the entry computation. 

Any reason why we want this inside while body, but not other fusible computations? For example, conditional body? For all fusible computations we have this helper: 
https://github.com/openxla/xla/blob/06e520756a5d9288d8b689fe387acab810bc3de8/xla/service/gpu/gpu_fusible.h#L81

Reply:
Thanks! I was looking for such a helper. While body just typically contains almost everything that is not in the entry computation.